addons/survey/tests/test_survey_results.py

```diff
@@ -0,0 +1,279 @@
+        line = self.UserInputLine.browse(user_input_line_id)
+        self.assertEqual(line.question_id.title,
+            'If a customer purchases a product on 6 January 2020, what is the latest day we expect to ship it?')
+        date_value = fields.Date.from_string('2020-01-08')
+        self.assertEqual(line.value_date, date_value)
+        lines_domain = [['&', ('value_date', '=', date_value), ('question_id', '=', line.question_id.id)]]
+        self._check_filters_results(post, self.certification_survey_id, lines_domain)
+
+    def test_results_page_filters_survey_mixed_models(self):
+        """
+            Test the survey results page filters with:
+                - a filter requesting the 'survey.question.answer' and 'survey.user_input.line' model
+                - resulting in one matching user_input record
+        """
+        post = {'filters': 'A,0,27|L,0,46'}
+
+        answer_id = 27
+        answer = self.QuestionAnswer.browse(answer_id)
+        self.assertEqual(answer.question_id.title, 'Select all the existing products')
+        self.assertEqual(answer.value, 'Drawer')
+        answer_line_domain = ['&', ('suggested_answer_id', '=', answer_id), ('question_id', '=', answer.question_id.id)]
+
+        user_input_line_id = 46
+        line = self.UserInputLine.browse(user_input_line_id)
+        self.assertEqual(line.question_id.title, 'Do you think we have missing products in our catalog? (not rated)')
+        self.assertEqual(line.value_text_box, 'I think it misses a product but I don\'t know what')
+        user_input_line_domain = ['&', ('value_text_box', '=', line.value_text_box), ('question_id', '=', line.question_id.id)]
+
+        self._check_filters_results(post, self.certification_survey_id, [answer_line_domain, user_input_line_domain])
+
+    def test_results_page_filters_survey_mixed_models_multiple(self):
+        """
+            Test the survey results page filters with:
+                - a filter requesting the 'survey.question.answer' and 'survey.user_input.line' model
+                - resulting in multiple matching user_input records
+        """
+        post = {'filters': 'A,0,41|L,0,53'}
+
+        answer_id = 41
+        answer = self.QuestionAnswer.browse(answer_id)
+        self.assertEqual(answer.question_id.title, 'How much do we sell our Cable Management Box?')
+        self.assertEqual(answer.value, '$100')
+        answer_line_domain = ['&', ('suggested_answer_id', '=', answer_id), ('question_id', '=', answer.question_id.id)]
+
+        user_input_line_id = 53
+        line = self.UserInputLine.browse(user_input_line_id)
+        self.assertEqual(line.question_id.title,
+            'If a customer purchases a product on 6 January 2020, what is the latest day we expect to ship it?')
+        self.assertEqual(line.value_date, fields.Date.from_string('2020-01-08'))
+        user_input_line_domain = ['&', ('value_date', '=', line.value_date), ('question_id', '=', line.question_id.id)]
```

Comment:
You have a chance to be more robust, future-proof using `expression.AND` as we do in the controller. If the details of the domain construction change, this could change with it.
```suggestion
        user_input_line_domain = expression.AND([('value_date', '=', line.value_date), ('question_id', '=', line.question_id.id)])
```